ee/backend/app/src/main/kotlin/io/tolgee/ee/api/v2/controllers/PromptController.kt

```diff
@@ -0,0 +1,162 @@
+  fun getAllPaged(
+    @ParameterObject pageable: Pageable,
+    @RequestParam search: String?,
+  ): PagedModel<PromptModel> {
+    val result = promptService.getAllPaged(projectHolder.project.id, pageable, search)
+    return arrayResourcesAssembler.toModel(result, promptModelAssembler)
+  }
+
+  @GetMapping("default")
+  @RequiresProjectPermissions([Scope.PROMPTS_VIEW])
+  fun getDefaultPrompt(): PromptDto {
+    return promptService.getDefaultPrompt()
+  }
+
+  @PostMapping("")
+  @RequiresProjectPermissions([Scope.PROMPTS_EDIT])
+  fun createPrompt(
+    @RequestBody @Valid dto: PromptDto,
+  ): PromptModel {
+    if (dto.template !== null) {
+      enabledFeaturesProvider.checkFeatureEnabled(
+        projectHolder.project.organizationOwnerId,
+        Feature.AI_PROMPT_CUSTOMIZATION,
+      )
+    }
+
+    val result = promptService.createPrompt(projectHolder.project.id, dto)
+    return promptModelAssembler.toModel(result)
+  }
+
+  @GetMapping("/{promptId}")
+  @RequiresProjectPermissions([Scope.PROMPTS_VIEW])
+  fun getPrompt(
+    @PathVariable promptId: Long,
+  ): PromptModel {
+    val result = promptService.findPrompt(projectHolder.project.id, promptId)
+    return promptModelAssembler.toModel(result)
+  }
+
+  @PutMapping("/{promptId}")
+  @RequiresProjectPermissions([Scope.PROMPTS_EDIT])
+  fun updatePrompt(
+    @PathVariable promptId: Long,
+    @RequestBody @Valid dto: PromptDto,
+  ): PromptModel {
+    if (dto.template !== null) {
+      enabledFeaturesProvider.checkFeatureEnabled(
+        projectHolder.project.organizationOwnerId,
+        Feature.AI_PROMPT_CUSTOMIZATION,
+      )
+    }
+    val result = promptService.updatePrompt(projectHolder.project.id, promptId, dto)
+    return promptModelAssembler.toModel(result)
+  }
+
+  @DeleteMapping("/{promptId}")
+  @RequiresProjectPermissions([Scope.PROMPTS_EDIT])
+  fun deletePrompt(
+    @PathVariable promptId: Long,
+  ) {
+    promptService.deletePrompt(projectHolder.project.id, promptId)
+  }
+
+  @PostMapping("run")
+  @RequiresProjectPermissions([Scope.PROMPTS_EDIT])
+  fun run(
+    @Valid @RequestBody data: PromptRunDto,
+  ): PromptResponseDto {
+    if (data.template !== null) {
+      enabledFeaturesProvider.checkFeatureEnabled(
+        projectHolder.project.organizationOwnerId,
+        Feature.AI_PROMPT_CUSTOMIZATION,
+      )
+    }
+
+    val projectId = projectHolder.project.id
+    val prompt =
+      promptService.getPrompt(
+        projectId,
+        data.template ?: promptDefaultService.getDefaultPrompt().template!!,
+        data.keyId,
+        data.targetLanguageId,
+        data.provider,
+        data.options,
+      )
+    val params = promptService.getLlmParamsFromPrompt(prompt, data.keyId, LlmProviderPriority.HIGH)
+    val organizationId = projectHolder.project.organizationOwnerId
+    val response =
+      promptService.runPrompt(
+        organizationId,
+        params,
+        data.provider
```

Comment:
Why do we identify provider by name and not by id?

Reply:
There can be multiple providers with the same name, then they are used for load-balancing. This will be explained in the documentation.